### PR TITLE
dashboard: Use local ajax over $.ajax; ajax: Create AjaxError

### PR DESCRIPTION
**Pull request by mc10**
I'm not too big of a fan of `ajax`'s error handling; any ideas for improvement?

### CODE REVIEW

File: lib/modules/dashboard.js [hunk 1]
```diff
@@ -793,8 +799,8 @@ function WidgetObject(widgetOptions) {
 			newSitetable($widgetContent[0]);
 		}
 	};
-	this.error = xhr => {
-		if (xhr.status === 404) {
+	this.error = e => {
+		if (e.message.startsWith('XHR status 404 ')) {
```
Comment (andytuba) [inline lib/modules/dashboard.js]: Dang, `e` doesn't include the xhr object?

Comment (mc10) [inline lib/modules/dashboard.js]: I don't think so; I think it's just the error object from
```js
throw new Error(`XHR status ${response.status} - url: ${url}`);
```

If there's a nicer approach I'm all ears.

**Comment by mc10** [comment]
Any reason why the Flow checking is failing? I don't think I wrote my `await ajax` call differently than the other call sites...

File: lib/modules/dashboard.js (hunk 1 above)
Comment (erikdesjardins) [inline lib/modules/dashboard.js]: We can throw a subclass of `Error`:

```js
class AjaxError extends Error {
    status: number;

    constructor(message: string, status: number) {
        super(message);
        this.status = status;
    }
}
```

**Comment by erikdesjardins** [comment]
FYI we're not merging 5.4.0 PRs right now (working towards 5.2.2), I suppose I shouldn't have approved this.

I'll revert if we do a 5.2.2 release (it may not happen).

**Comment by mc10** [comment]
My bad, I didn't notice the tag. Feel free to revert if you'd like.

**Pull request merged**